internal/presentation/presentation.go

```diff
@@ -126,6 +131,118 @@ func (e Output) undefined() bool {
 	return len(e.Result) == 0 && (e.Partial == nil || len(e.Partial.Queries) == 0)
 }
 
+// NewOutputErrors creates a new slice of OutputError's based
+// on the type of error passed in. Known structured types will
+// be translated as appropriate, while unknown errors are
+// placed into a structured format with their string value.
+func NewOutputErrors(err error) []OutputError {
+	var errs []OutputError
+	if err != nil {
+		// Handle known structured errors
+
+		switch typedErr := err.(type) {
+		case *ast.Error:
+			oe := OutputError{
```

Comment:
I noticed through manual testing that parser errors are emitting an empty details object. For example:

```
$ opa eval -d ~/x.json '!!!'
{
  "errors": [
    {
      "message": "no match found",
      "type": "ast",
      "code": "rego_parse_error",
      "location": {
        "file": "",
        "row": 1,
        "col": 1
      },
      "details": {}
    }
  ]
}
```

We use parserErrorDetail to output the raw line text with a carat. One option would be to export the detail struct. Thoughts?

Reply:
Hmm, I'll have to look into that. Making it public might help, will have to check and see why some of the other details structs work (one of the unit tests ast errors has details).

Reply:
The other detail structs are exported and have JSON tags on them.

Reply:
Updated it to be exported and have tags. I checked for any others like that in `ast` and didn't see any.